libqtile/bar.py

```diff
@@ -151,14 +157,20 @@ def __init__(self, widgets, size, **config):
         self.queued_draws = 0
 
     def _configure(self, qtile, screen):
-        if self not in [screen.top, screen.bottom]:
-            raise confreader.ConfigError(
-                "Bars must be at the top or the bottom of the screen."
-            )
-        if len([w for w in self.widgets if w.width_type == STRETCH]) > 1:
+        Gap._configure(self, qtile, screen)
+
+        stretches = 0
+        for w in self.widgets:
+            # Executing _test_orientation_compatibility later, for example in
```

Comment:
Can we do this in __init__? _configure is called every time a screen is resized, and I don't think we need to do it more than once.